### PR TITLE
705_inform_the_counterparty

### PR DESCRIPTION
**Pull request by sf316**
# Description

Following the same implementation made in PR #726, I decided to handle successful executions in the same place. As a result, filenames have been changed. To view the changes made in the event handler, please check the very first commit. Also, I think placeholder should be put inside the loop. Please let me know if there are any errors or concerns.

## Related Issue

#705 
#703 
#726 

## How Has This Been Tested

Needs tests later.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Request to be added as a Code Owner/Maintainer

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I commit to abide by the Responsibilities of Code Owners/Maintainers.

### CODE REVIEW

**Review by Therecanbeonlyone1969 — APPROVED** [review]
LGTM

File: examples/bri-3/src/bri/vsm/capabilites/executeVsmCycle/executeVsmCycleCommand.handler.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { TransactionAgent } from '../../../transactions/agents/transactions.agen
 import { TransactionStatus } from '../../../transactions/models/transactionStatus.enum';
 import { WorkstepStorageAgent } from '../../../workgroup/worksteps/agents/workstepsStorage.agent';
 import { ExecuteVsmCycleCommand } from './executeVsmCycle.command';
-import { WorkstepExecutionFailuresEvent } from '../handleWorkstepFailuresEvents/workstepExecutionFailures.event';
+import { WorkstepExecutionEvent } from '../handleWorkstepEvents/workstepExecution.event';
```
Comment (ognjenkurtic) [inline examples/bri-3/src/bri/vsm/capabilites/executeVsmCycle/executeVsmCycleCommand.handler.ts:7]: i think we can adopt naming for events in past tense, as they mark that something has  happened:

WorkstepExecutionEvent ->  WorkstepExecutedEvent

File: examples/bri-3/src/bri/vsm/capabilites/executeVsmCycle/executeVsmCycleCommand.handler.ts [hunk 2]
```diff
@@ -48,12 +48,14 @@ export class ExecuteVsmCycleCommandHandler
...
         this.txStorageAgent.updateTransactionStatus(tx);
       } catch (error) {
-        this.eventBus.publish(new WorkstepExecutionFailuresEvent(tx, error));
+        this.eventBus.publish(new WorkstepExecutionEvent(tx, error));
         tx.updateStatusToAborted();
         this.txStorageAgent.updateTransactionStatus(tx);
         return;
       }
+
+      // TODO: #705 Relevant BPI subjects are informed (Notification is thrown?)
```
Comment (ognjenkurtic) [inline examples/bri-3/src/bri/vsm/capabilites/executeVsmCycle/executeVsmCycleCommand.handler.ts:57]: please remove TODO

File: examples/bri-3/src/bri/vsm/capabilites/handleWorkstepEvents/workstepExecution.handler.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+  private constructPayload(message: string) {
+    return {
+      id: 'xxx',
+      info: message,
+    };
+  }
+
+  private async publishMessage(channels: string[], bpiMessage: BpiMessage) {
+    await Promise.all(
```
Comment (ognjenkurtic) [inline examples/bri-3/src/bri/vsm/capabilites/handleWorkstepEvents/workstepExecution.handler.ts:63]: i do not think we need the promise.all construct  here. just iterate channels and trigger publishing as already done in the loop. The method return type can  be Promise<  void >

File: examples/bri-3/src/bri/vsm/capabilites/handleWorkstepEvents/workstepExecution.handler.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    );
+
+    // Change channels to publish message here
+    const channels = [event.tx.fromBpiSubjectAccount.ownerBpiSubject.publicKey];
+
+    if (event.status == 'Success') {
+      channels.push(event.tx.toBpiSubjectAccount.ownerBpiSubject.publicKey);
+    }
+
+    this.publishMessage(channels, bpiMessage);
```
Comment (ognjenkurtic) [inline examples/bri-3/src/bri/vsm/capabilites/handleWorkstepEvents/workstepExecution.handler.ts:52]: missing  await

File: examples/bri-3/src/bri/vsm/vsm.module.ts [hunk 5]
```diff
@@ -7,12 +7,12 @@ import { WorkstepModule } from '../workgroup/worksteps/worksteps.module';
 import { VsmTasksSchedulerAgent } from './agents/vsmTaskScheduler.agent';
 import { ExecuteVsmCycleCommandHandler } from './capabilites/executeVsmCycle/executeVsmCycleCommand.handler';
 import { MessagingAgent } from '../communication/agents/messaging.agent';
-import { WorkstepExecutionFailuresHandler } from './capabilites/handleWorkstepFailuresEvents/workstepExecutionFailures.handler';
+import { WorkstepExecutionHandler } from './capabilites/handleWorkstepEvents/workstepExecution.handler';
```
Comment (ognjenkurtic) [inline examples/bri-3/src/bri/vsm/vsm.module.ts:10]: WorkstepExecutedEventHandler

File: examples/bri-3/src/bri/vsm/capabilites/handleWorkstepEvents/workstepExecution.handler.ts
Comment (ognjenkurtic) [inline examples/bri-3/src/bri/vsm/capabilites/handleWorkstepEvents/workstepExecution.handler.ts:1]: rename file to workstepExecutedEvent.handler.ts

File: examples/bri-3/src/bri/vsm/capabilites/handleWorkstepEvents/workstepExecution.event.ts
Comment (ognjenkurtic) [inline examples/bri-3/src/bri/vsm/capabilites/handleWorkstepEvents/workstepExecution.event.ts:1]: workstepExecuted.event.ts

File: examples/bri-3/src/bri/vsm/capabilites/executeVsmCycle/executeVsmCycleCommand.handler.ts (hunk 1 above)
Comment (sf316) [reply examples/bri-3/src/bri/vsm/capabilites/executeVsmCycle/executeVsmCycleCommand.handler.ts]: @ognjenkurtic Done. Thank you for reviewing!

**Review by ognjenkurtic — DISMISSED** [review]

File: examples/bri-3/src/bri/vsm/capabilites/handleWorkstepEvents/workstepExecutedEvent.handler.ts [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    if (event.status == 'Success') {
+      channels.push(event.tx.toBpiSubjectAccount.ownerBpiSubject.publicKey);
+    }
+
+    await this.publishMessage(channels, bpiMessage);
+  }
+
+  private constructPayload(message: string) {
+    return {
+      id: 'xxx',
```
Comment (Kasshern) [inline examples/bri-3/src/bri/vsm/capabilites/handleWorkstepEvents/workstepExecutedEvent.handler.ts:57]: @sf316 what is this id meant to be? please add todo or replace with v4() if jsut in need of uuid.

**Review by Kasshern — COMMENTED** [review]
Just left a single comment. Otherwise looks good. Will approve after its answered.

File: examples/bri-3/src/bri/vsm/capabilites/handleWorkstepEvents/workstepExecutedEvent.handler.ts (hunk 6 above)
Comment (sf316) [reply examples/bri-3/src/bri/vsm/capabilites/handleWorkstepEvents/workstepExecutedEvent.handler.ts]: @Kasshern Add TODO. Thank you for reviewing!

**Review by Therecanbeonlyone1969 — APPROVED** [review]
LGTM now

**Review by ognjenkurtic — APPROVED** [review]

**Pull request merged**